### PR TITLE
[feat]: [CDS-45159]: add validation fields between entity yaml & crud request

### PR DESCRIPTION
**Pull request by tchaurasiya**
… request

## Describe your changes

We are adding validation to compare basic fields [identifier,name,orgId,projectId,env/infra type] between request and yaml.

## Checklist
- [x] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/42502)

### CODE REVIEW

**Review by yogesh-chauhan — CHANGES_REQUESTED** [review]
> Reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42502)* status: 3 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`127-cd-nextgen-entities/src/main/java/io/harness/cdng/infra/mapper/InfrastructureEntityConfigMapper.java` line 94 at r1](https://reviewable.io/reviews/harness/harness-core/42502#-NLW6bAyAGIDxWf2fsHI:-NLW6bAyAGIDxWf2fsHJ:b-q8ikkz) ([raw file](https://github.com/harness/harness-core/blob/88895fa4f6bf42451795c5bfbe9d7c74e8fa257c/127-cd-nextgen-entities/src/main/java/io/harness/cdng/infra/mapper/InfrastructureEntityConfigMapper.java#L94)):*
> ```Java
>         != 0) {
>       throw new InvalidRequestException(
>           String.format("Org Identifier %s passed in yaml is not same as passed in query params %s",
> ```

will this always happen from query params ?

File: 127-cd-nextgen-entities/src/main/java/io/harness/cdng/infra/mapper/InfrastructureEntityConfigMapper.java [hunk 1]
```diff
@@ -83,4 +93,37 @@ public List<InfrastructurePlanCreatorConfig> toInfrastructurePlanCreatorConfig(
                    .build())
         .collect(Collectors.toList());
   }
+
+  private static void validateFieldsOrThrow(InfrastructureConfig config, InfrastructureEntity entity) {
```
Comment (yogesh-chauhan) [inline 127-cd-nextgen-entities/src/main/java/io/harness/cdng/infra/mapper/InfrastructureEntityConfigMapper.java:97]: config -> fromYaml

entity -> requestedEntity

**Review by yogesh-chauhan — CHANGES_REQUESTED** [review]

**Review by yogesh-chauhan — APPROVED** [review]
> Reviewed 11 of 11 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42502)* status: 12 file(s) to review. 1 of 2 approvals obtained.

**Review by hingerabhinav — APPROVED** [review]
> Reviewed 1 of 1 files at r3, 11 of 11 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/42502)* status: :shipit: complete! ✓

**Comment by tchaurasiya** [comment]
trigger messagecheck

**Pull request merged**